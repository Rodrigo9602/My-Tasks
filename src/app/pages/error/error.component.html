<div class="notfound-wrapper">
    <div class="notfound">    
        <h1>Oops!</h1>
        <h2>404 - The Page can't be found</h2>    
    <a class="btn-link" [routerLink]="['/home']">Go to Homepage</a>
</div>
</div>
