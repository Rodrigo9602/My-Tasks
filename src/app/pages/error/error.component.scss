
.notfound-wrapper {
  width: 100vw;
  height: 100vh;
  background-image: url('../../../assets/images/error-bg.jpg');
  background-repeat: no-repeat;
  background-size: cover;
  display: flex;    
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
}

.notfound {
  width: 80vw;
  height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center; 
  
  h1 {
    font-family: 'poppins';
    font-size: 12rem;
    color: var(--items-color-gradient-sec);
  }
  h2 {
    font-family: 'poppins';
    font-size: 24px;
    margin-top: -100px;
  }

  .btn-link {
    width: 13%;
    height: 40px;
    background-color: var(--items-color-gradient);
    color: #fff;
    text-transform: uppercase;
    border: 1px solid transparent;
    border-radius: 0.5rem;
    outline: none;
    display: flex;
    align-items: center;
    justify-content: center;
    cursor: pointer;
  }
}

