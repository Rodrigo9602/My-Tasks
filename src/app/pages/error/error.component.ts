import { Component } from '@angular/core';
import { RouterModule } from '@angular/router';

@Component({
  selector: 'app-error',
  standalone: true,
  imports: [RouterModule],
  templateUrl: './error.component.html',
  styleUrl: './error.component.scss'
})
export class ErrorComponent {

}
