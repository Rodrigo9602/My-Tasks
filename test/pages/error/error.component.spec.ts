import { ComponentFixture, TestBed } from '@angular/core/testing';

import { ErrorComponent } from '../../../src/app/pages/error/error.component';
import { ActivatedRoute } from '@angular/router';


describe('ErrorComponent', () => {
  let component: ErrorComponent;
  let fixture: ComponentFixture<ErrorComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [ErrorComponent],
      providers: [
        { provide: ActivatedRoute, useValue: {snapshot: {}}} 
      ]
    })
    .compileComponents();
    
    fixture = TestBed.createComponent(ErrorComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match initial snapshot', () => {
    expect(compiled).toMatchSnapshot();
  });
});
