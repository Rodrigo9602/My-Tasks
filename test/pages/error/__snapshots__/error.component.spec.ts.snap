// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorComponent should match initial snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="notfound-wrapper"
  >
    <div
      class="notfound"
    >
      <h1>
        Oops!
      </h1>
      <h2>
        404 - The Page can't be found
      </h2>
      <a
        class="btn-link"
        href="/home"
      >
        Go to Homepage
      </a>
    </div>
  </div>
</div>
`;
